[RTI-2784] Maintenance(grid): allow all console methods (except for console.error) in example generator

sdwvit opened the pull request:
- Introduced an array to match more console methods for better detection in example gen.
 - Updated getHasExampleConsoleLog to utilize the new logic.
 - Still do not allow console.error, as it is reserved for service messages like absent licence error
 - Update extreme pivot example to use console.warn

| Was | Now |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/f19b81ef-efd0-4faf-9b16-6111f24bbd84)  | ![Screenshot 2025-06-23 at 16 08 49](https://github.com/user-attachments/assets/2c46facc-a3d0-4a14-af1c-baa8f7520755)  | 

Also fixes 
https://ag-grid.atlassian.net/jira/software/projects/RTI/boards/23?selectedIssue=RTI-2824

| Was | Now |
|--------|--------|
| ![Screenshot 2025-06-23 at 16 24 10](https://github.com/user-attachments/assets/25e5bd0c-3b63-4a3e-8d65-773fdc4d7a80) | ![Screenshot 2025-06-23 at 16 22 47](https://github.com/user-attachments/assets/1375591e-00a9-4dbf-880f-c043448d4b1f)  |

sdwvit committed:
Enhance console log detection in fileUtils

 - Introduced a regex to match all console methods for better detection.
 - Updated getHasExampleConsoleLog to utilize the new regex.
 - Reset lastIndex on the regex to ensure it can be reused.

sdwvit committed:
Unnecessary escape

sdwvit committed:
- adjust generator to patch all console methods except for console.error, since that is used for license info printout
- change extreme-pivot log to level warn

taktran commented on plugins/ag-grid-generate-example-files/src/executors/generate/generator/utils/getConsoleLogSnippet.ts:
```diff
@@ -161,21 +161,23 @@ function patchConsoleLog() {
-        }
-        originalConsoleLog(...args);
-    };
+    // we ignore console.error to avoid printing license messages
```
It's also so that there isn't an infinite loop if the try/catch fails

sdwvit replied:
yep

taktran commented on plugins/ag-grid-generate-example-files/src/executors/generate/generator/utils/fileUtils.ts:
```diff
@@ -174,9 +174,13 @@ export function convertTsxToJsx(fileStr: string): string {
     return jsxFile;
 }
+const consoleMethods = Object.getOwnPropertyNames(console);
+const consoleRegexp = new RegExp(`console.(${consoleMethods.join('|')})`, 'g');
```
I don't think we want to support all of what `console` supports, as the parsing logic would need to change as well. An allow list of `log` and `warn` should be sufficient.

sdwvit committed:
- Update consoleMethods to explicitly list allowed methods.
 - Modify console log patching to include additional methods.
 - Clarify comments regarding the exclusion of console.error.

taktran reviewed (approved):
Good stuff 👍🏼
